cpp/include/rapids_triton/triton/responses.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <triton/core/tritonbackend.h>
+#include <algorithm>
+#include <iterator>
+#include <rapids_triton/exceptions.hpp>
+#include <rapids_triton/triton/logging.hpp>
+#include <vector>
+
+namespace triton {
+namespace backend {
+namespace rapids {
+
+template <typename Iter>
+auto construct_responses(Iter requests_begin, Iter requests_end)
+{
+  auto responses = std::vector<TRITONBACKEND_Response*>{};
+
+  auto requests_size = std::distance(requests_begin, requests_end);
+  if (!(requests_size > 0)) {
```

Comment:
```suggestion
  if (requests_size <= 0) {
```

Reply:
I have a slight preference for leaving this as is. The semantic intent of this is that we're looking for something **non**-positive and **non**-zero as opposed to that we're looking for something negative. Okay by you if we leave it as is?

Reply:
Yes no problem